src/mplfinance/_utils.py

```diff
@@ -81,6 +83,49 @@ def roundTime(dt=None, roundTo=60):
    rounding = (seconds+roundTo/2) // roundTo * roundTo
    return dt + datetime.timedelta(0,rounding-seconds,-dt.microsecond)
 
+def _calculate_atr(atr_length, highs, lows, closes):
+    """Calculate the average true range
+    atr_length : time period to calculate over
+    all_highs : list of highs
+    all_lows : list of lows
+    all_closes : list of closes
+    """
+    if atr_length < 0:
+        raise ValueError("Specified atr_length may not be less than 0")
+    elif atr_length > len(closes):
```

Comment:
I am wondering if this needs to be `elif atr_length`**`>=`**` len(closes):`??  

The reason is, when `atr_length == len(closes)`, then the first time through the loop below `close_prev` is actually the **last close** in the series (because `closes[i-1]` for i=0 references the ***last*** item in the list).

This means that the "true range" (on the first iteration) is calculated by comparing the *high, low, and close of the first data point, with the close of the last data point.*  

I don't know if this is correct or not, with regard to what the "average true range" calculation is supposed to be, but it can skew the atr, because it can result in a **tr** value that is very different from all of the other tr values.

Reply:
Yes, you are correct.  The code would end up using the last close for the first true value and this isn't what the tr calculation wants. I'll change this in my latest version.